src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java

```diff
@@ -79,8 +80,16 @@ public PartitionIterator filterReplicaFilteringProtection(PartitionIterator full
     {
         for (RowFilter.Expression expression : controller.filterOperation())
         {
-            if (controller.getContext(expression).getAnalyzer().transformValue())
-                return applyIndexFilter(fullResponse, analyzeFilter(), queryContext);
+            AbstractAnalyzer analyzer = controller.getContext(expression).getAnalyzerFactory().get();
+            try
+            {
+                if (analyzer.transformValue())
+                    return applyIndexFilter(fullResponse, analyzeFilter(), queryContext);
+            }
+            finally
+            {
+                analyzer.end();
```

Comment:
BTW: (very minor) why does it have to be `analyzer.end()` and not `close()` from `AutoCloseable` ? Then you could just use a shorter try with resources.

Reply:
Then it throws an exception when an exception at that stage is probably close to impossible.

Reply:
The fact that an exception is close to impossible does not mean we are allowed to wrap into unchecked. 
The decision between checked / unchecked is based on the recoverability of the problem, not its likelihood.
If you meant "the exception can be thrown here only as a result of a bug in the code, not as a result of user input or environment", then I agree, unchecked is the way to go and `AutoCloseable` is not the right fit.